### PR TITLE
Head, Hips and Chest IK can now be controlled from Controller inputs

### PR DESCRIPTION
**Pull request by hyperlogic**
Previously the MyAvatar.addAnimationStateHandler() was the only mechanism to control the avatar's IK targets for the head, hips and chest a.k.a spine2.  Now the input system supports standard poses for the Head, Hips and Spine2 and routes those poses correctly to the underlying IK system.

### CODE REVIEW

**Comment by hyperlogic** [comment]
QA Test:
   * run the following script scripts/developer/tests/hipsControllerTest.js
   * press both triggers on your hand controllers
   * this should cause your hips to gyrate.  This scripts uses the new Standard.Hips input device to control the IK system's hip target.
   * Pressing the triggers again should cause your hips to return to normal.

File: interface/src/avatar/MyAvatar.cpp [hunk 1]
```diff
@@ -2599,6 +2664,79 @@ glm::vec3 MyAvatar::getAbsoluteJointTranslationInObjectFrame(int index) const {
     }
 }
 
+glm::mat4 MyAvatar::getCenterEyeCalibrationMat() const {
+    // TODO: as an optimization cache this computation, then invalidate the cache when the avatar model is changed.
+    int rightEyeIndex = _rig->indexOfJoint("RightEye");
+    int leftEyeIndex = _rig->indexOfJoint("LeftEye");
+    if (rightEyeIndex >= 0 && leftEyeIndex >= 0) {
+        auto centerEyePos = (getAbsoluteDefaultJointTranslationInObjectFrame(rightEyeIndex) + getAbsoluteDefaultJointTranslationInObjectFrame(leftEyeIndex)) * 0.5f;
```
Comment (sethalves) [inline interface/src/avatar/MyAvatar.cpp]: Isn't the middle eye usually slightly higher on the Y-axis than the other two?  see http://thespiritscience.net/wp-content/uploads/2015/06/ThirdEye.jpg

Comment (hyperlogic) [inline interface/src/avatar/MyAvatar.cpp]: 👁 👁 👁

**Comment by hyperlogic** [comment]
build failed due to a git login error?  Trying again:  Test this please.

**Comment by hifi-gustavo** [comment]
Build console output for commit: f3cc1c2d86d7bcb5e568ac3dca7bedd6544ab906 available at: https://gist.github.com/5432a0d917c71a9551ec3efba956d87e<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10319-f3cc1c2.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10319-f3cc1c2.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: f3cc1c2d86d7bcb5e568ac3dca7bedd6544ab906 available at: https://gist.github.com/5432a0d917c71a9551ec3efba956d87e<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10319-f3cc1c2.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10319-f3cc1c2.dmg

**Comment by hyperlogic** [comment]
jenkins error:

    Caused by: hudson.plugins.git.GitException: Command "git -c core.askpass=true fetch --no-tags --progress https://github.com/highfidelity/hifi.git +refs/pull/10319/*:refs/remotes/origin/pr/10319/* --depth=1" returned status code 143:

try again, test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: 58de67ba119268b45280b4a0fad83b91183e2d9b available at: https://gist.github.com/5432a0d917c71a9551ec3efba956d87e<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10319-58de67b.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10319-58de67b.dmg

**Comment by hyperlogic** [comment]
bugsplat fail upload on windows, and macosx fail during instal of all in one build?

   WARNING: No response from www.bugsplatsoftware.com/browse/symbol_upload3.php

Try again, test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: 58de67ba119268b45280b4a0fad83b91183e2d9b available at: https://gist.github.com/5432a0d917c71a9551ec3efba956d87e<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10319-58de67b.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10319-58de67b.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 58de67ba119268b45280b4a0fad83b91183e2d9b available at: https://gist.github.com/5432a0d917c71a9551ec3efba956d87e<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10319-58de67b.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10319-58de67b.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 0c09823f6731e15e50c5bf361144ea120fb94d20 available at: https://gist.github.com/5432a0d917c71a9551ec3efba956d87e<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10319-0c09823.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10319-0c09823.dmg

**Comment by hyperlogic** [comment]
bugsplat upload failure!  try again test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: 0c09823f6731e15e50c5bf361144ea120fb94d20 available at: https://gist.github.com/5432a0d917c71a9551ec3efba956d87e<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10319-0c09823.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10319-0c09823.dmg

**Comment by claudioshg** [comment]
testing

**Comment by claudioshg** [comment]
### Desktop PC - Windows 10 Home - CPU 4.0GHz Intel i7 - GTX1070 - RAM 16GB - Tested with Oculus Rift

Tested: Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10319-0c09823.exe

PASS: Pressing both triggers starts the gyration, and pressing both triggers again stops it.

hifi-snap-by--on-2017-05-01_11-36-50

**Pull request merged**